lib/src/Uncertainty/Distribution/ParetoFactory.cxx

```diff
@@ -0,0 +1,120 @@
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/ParetoFactory.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/MaximumLikelihoodFactory.hxx"
+#include "openturns/MethodOfMomentsFactory.hxx"
+#include "openturns/Brent.hxx"
+#include "openturns/SymbolicFunction.hxx"
+#include "openturns/LeastSquaresFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(ParetoFactory)
+
+static const Factory<ParetoFactory> Factory_ParetoFactory;
+
+/* Default constructor */
+ParetoFactory::ParetoFactory()
+  : DistributionFactoryImplementation()
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+ParetoFactory * ParetoFactory::clone() const
+{
+  return new ParetoFactory(*this);
+}
+
+/* Here is the interface that all derived class must implement */
+
+Distribution ParetoFactory::build(const Sample & sample) const
+{
+  return buildAsPareto(sample).clone();
+}
+
+Distribution ParetoFactory::build(const Point & parameters) const
+{
+  return buildAsPareto(parameters).clone();
+}
+
+Distribution ParetoFactory::build() const
+{
+  return buildAsPareto().clone();
+}
+
+Pareto ParetoFactory::buildMethodOfMoments(const Sample & sample) const
+{
+  // numerically inverse the skewness
+  SymbolicFunction constraint("alpha", OSS() << "2*(1+alpha)/(alpha-3)*sqrt((alpha-2)/alpha)");
+  const Scalar skewness = sample.computeSkewness()[0];
+  if (skewness < constraint(Point(1, ResourceMap::GetAsScalar("ParetoFactory-AlphaUpperBound")))[0])
+    throw InvalidArgumentException(HERE) << "alpha is not defined";
+  const Brent solver;
+  // alpha> 3.0 so as the skewness to be defined
+  const Scalar alpha = solver.solve(constraint, skewness, 3.0 + std::sqrt(SpecFunc::Precision), ResourceMap::GetAsScalar("ParetoFactory-AlphaUpperBound"));
+  const Scalar beta = (alpha - 1.0) * std::sqrt((alpha - 2.0) / alpha) * sample.computeStandardDeviationPerComponent()[0];
+  const Scalar gamma = sample.computeMean()[0] - alpha * beta / (alpha - 1.0);
+
+  Pareto result(beta, alpha, gamma);
+  result.setDescription(sample.getDescription());
+  return result;
+}
+
+Pareto ParetoFactory::buildMethodOfLikelihoodMaximization(const Sample & sample) const
+{
+  const MaximumLikelihoodFactory factory(buildMethodOfMoments(sample));
+  return buildAsPareto(factory.build(sample).getParameter());
+}
+
+Pareto ParetoFactory::buildMethodOfLeastSquares(const Sample & sample) const
+{
+  const LeastSquaresFactory factory(buildMethodOfMoments(sample));
```

Comment:
IMHO we should have similar procedure as for `LogNormalFactory:: buildMethodOfLeastSquares`

For a fixed `gamma`, we derive `alpha` and `beta` coefficients. Residual is difference between empirical `log-survival function` and one issued from distribution (linear regression)